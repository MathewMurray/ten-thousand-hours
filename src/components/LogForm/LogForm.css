.LogForm select, .ReviewForm label {
    font-size: 1.1em;
  }
  
  .LogForm label {
    display: inline-block;
    margin: 10px 10px 20px;
    margin-left: 0;
    color: #e0e1dd;

  }