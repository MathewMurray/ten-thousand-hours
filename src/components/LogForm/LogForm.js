import React, {Component} from 'react'
import GoalContext from '../../Context/GoalsContext'
import GoalApiService from '../../services/goals-api-service'
import {Button,Textarea} from '../Utils/Utils'
import './LogForm.css'

export default class LogForm extends Component {
    static contextType = GoalContext

    handleSubmit = ev => {
        ev.preventDefault()
        const{goal} = this.context
        const {text,user_hours} = ev.target

        GoalApiService.postLog(goal.id,text.value,Number(user_hours.value))
            .then(this.context.addLog)
            .then(() => {
                text.value = ''
            })
            .catch(this.context.setError)
    }

    render(){
        return (
            <form className='LogForm' onSubmit={this.handleSubmit}>
                <div className='text'>
                    <Textarea
                        required
                        aria-label='Type a log...'
                        name='text'
                        id='text'
                        col='30'
                        rows='3'
                        placeholder='Type a log..'>
                    </Textarea>
                </div>
                <div className='user_hours'>
                    <label htmlFor='user_hours'>How many hours?</label>
                    <Textarea
                        required
                        aria-label='Log your time..'
                        name='user_hours'
                        id='user_hours'
                        placeholder='1,2,3...'>
                    </Textarea>
                </div>
                <Button type='submit'>
                    Post Log
                </Button>
            </form>
        )
    }
}