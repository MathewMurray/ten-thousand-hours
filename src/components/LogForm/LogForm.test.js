import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import LogForm from './LogForm'

describe(`LogForm component`, () => {
    it('renders the log form', () => {
        const wrapper = shallow(<LogForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})