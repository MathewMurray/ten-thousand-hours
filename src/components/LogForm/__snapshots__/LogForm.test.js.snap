// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogForm component renders the log form 1`] = `
<form
  className="LogForm"
  onSubmit={[Function]}
>
  <div
    className="text"
  >
    <Textarea
      aria-label="Type a log..."
      col="30"
      id="text"
      name="text"
      placeholder="Type a log.."
      required={true}
      rows="3"
    />
  </div>
  <div
    className="user_hours"
  >
    <label
      htmlFor="user_hours"
    >
      How many hours?
    </label>
    <Textarea
      aria-label="Log your time.."
      id="user_hours"
      name="user_hours"
      placeholder="1,2,3..."
      required={true}
    />
  </div>
  <Button
    type="submit"
  >
    Post Log
  </Button>
</form>
`;
